Source/ACE/Entity/Player.cs

```diff
@@ -765,15 +770,96 @@ public void SaveOptions()
+        /// </summary>
+        public void SetPortalRecallCharacterPosition(CharacterPosition newPosition)
+        {
+            // TODO : Save the Portal Recall position information from the destination of a portal
+            character.SetCharacterPositions(CharacterPositionType.PortalRecall, newPosition);
+        }
+
+        /// <summary>
+        /// Set's the Primary tied portal CharacterPosition for use with the portal recall spells
+        /// </summary>
+        public void SetPrimaryPortalRecallCharacterPosition(CharacterPosition newPosition)
+        {
+            // TODO : Save the Portal Recall position information from the destination of a portal
+            character.SetCharacterPositions(CharacterPositionType.PrimaryPortalRecall, newPosition);
+        }
+
+        /// <summary>
+        /// Set's the Secondary tied portal CharacterPosition for use with the portal recall spells
+        /// </summary>
+        public void SetSecondaryPortalRecallCharacterPosition(CharacterPosition newPosition)
+        {
+            // TODO : Save the Portal Recall position information from the destination of a portal
+            character.SetCharacterPositions(CharacterPositionType.SecondaryPortalRecall, newPosition);
+        }
+
+        /// <summary>
+        /// Set's allegiance recall position
+        /// </summary>
+        public void SetAllegianceCharacterPosition(CharacterPosition newPosition)
+        {
+            // TODO : Save the Allegiance Recall position information from the destination of a bind stone
+            character.SetCharacterPositions(CharacterPositionType.AllegianceHometown, newPosition);
+        }
+
+        /// <summary>
+        /// Set's mansion recall position
+        /// </summary>
+        public void SetMansionCharacterPosition(CharacterPosition newPosition)
+        {
+            // TODO : Save the Mansion Recall position information from the destination of a house box
+            character.SetCharacterPositions(CharacterPositionType.MansionRecall, newPosition);
+        }
+
+        /// <summary>
+        /// Saves a CharacterPosition to the character position dictionary
+        /// </summary>
+        /// <param name="type"></param>
+        /// <param name="newPosition"></param>
+        public void SetCharacterPosition(CharacterPositionType type, CharacterPosition newPosition)
+        {
+            character.SetCharacterPositions(type, newPosition);
+        }
+
+        /// <summary>
         /// Saves the character to the persistent database. Includes Stats, Position, Skills, etc.
         /// </summary>
         public void SaveCharacter()
         {
             if (character != null)
             {
-                // save the player position to the database blob
-                character.Position = Session.Player.Position;
-
+                // Save the current position to persistent storage
+                SetPhysicalCharacterPosition();

```

Comment:
Loads the Player position into the dictionary of positions before saving in the database.